src/platform/plugins/shared/discover/public/application/main/components/chart/use_discover_histogram.ts

```diff
@@ -79,8 +80,11 @@ export const useDiscoverHistogram = (
   const updateAppState = useCurrentTabAction(internalStateActions.updateAppState);
   const documentsState = useDataState(documents$);
   const isChartLoading = useMemo(() => {
-    return isEsqlMode && documentsState?.fetchStatus === FetchStatus.LOADING;
-  }, [isEsqlMode, documentsState?.fetchStatus]);
+    const query = stateContainer.getCurrentTab().appState.query;
+    if (!isOfAggregateQueryType(query)) return false;
+    if (!hasTransformationalCommand(query.esql)) return false;
+    return documentsState?.fetchStatus === FetchStatus.LOADING;
+  }, [documentsState?.fetchStatus, stateContainer]);
```

Comment:
```ts
  const query = useAppStateSelector((state) => state.query);
  const isChartLoading = useMemo(() => {
    if (!isOfAggregateQueryType(query)) return false;
    if (!hasTransformationalCommand(query.esql)) return false;
    return documentsState?.fetchStatus === FetchStatus.LOADING;
  }, [documentsState?.fetchStatus, query]);
```
Nit: probably doesn't have an actual impact at runtime, but ideally we'd use a selector for `query` so it's reactive.